process/block/metablock.go

```diff
@@ -1900,18 +1945,36 @@ func (mp *metaProcessor) receivedShardHeader(headerHandler data.HeaderHandler, s
 			}
 		}
 
+		if hdrInfoForHash != nil && !hdrInfoForHash.hasProof {
+			hasProof := mp.proofsPool.HasProof(shardHeader.GetShardID(), shardHeaderHash)
+			hdrInfoForHash.hasProof = hasProof
+			if hasProof {
+				mp.hdrsForCurrBlock.missingHdrsProofs--
+			}
+		}
+
 		if mp.hdrsForCurrBlock.missingHdrs == 0 {
-			mp.hdrsForCurrBlock.missingFinalityAttestingHdrs = mp.requestMissingFinalityAttestingShardHeaders()
+			if !mp.shouldConsiderProofsForNotarization(shardHeader) {
```

Comment:
on the else, you should check that there is a valid proof for this header, otherwise add it to the missing proofs.